lib/teleterm/daemon/gateway_cert_reissuer.go

```diff
@@ -43,12 +43,14 @@ type GatewayCertReissuer struct {
 	Log       *logrus.Entry
 }
 
-// DBCertReissuer lets us pass a mock in tests and clusters.Cluster (which makes calls to the
+// CertReissuer lets us pass a mock in tests and clusters.Cluster (which makes calls to the
 // cluster) in production code.
-type DBCertReissuer interface {
+type CertReissuer interface {
 	// ReissueDBCerts reaches out to the cluster to get a cert for the specific tlsca.RouteToDatabase
 	// and saves it to disk.
 	ReissueDBCerts(context.Context, tlsca.RouteToDatabase) error
+	// TODO
+	ReissueKubeCerts(context.Context, string) error
```

Comment:
Hmm, I'm not sure what do to do about this one.

`CertReissuer` was added mostly to mock out in tests parts of the code which deal with actual certs.

From the looks of it, a better design might be to have a single method such as `ReissueCerts(context.Context, *gateway.Gateway)` and pass a different `CertReissuer` to `GatewayCertReissuer` when the gateway is created.

The problem is that there's a single `GatewayCertReissuer` created for the whole `daemon.Service`, so this would have to be refactored first.

If you don't see a clear way to address this, I think it'd be fine to leave it as is and reconsider the refactor when we'll be adding a third gateway type.

I'm still not 100% sure how to best models this kind of relationship in Go, I don't know if it wants to follow OOP or FP more in this regard. With how interfaces work, I'd have assumed that it should be more similar to OOP, which is what I described earlier in the comment.